### ISSUE TITLE
append mode fails if a directory already exists

### ISSUE THREAD

**Issue by BrynCooke** [iomentum/cargo-scaffold#14]
Using the following:
```
            let opts = cargo_scaffold::Opts::builder()
                .template_path("scaffold/plugin")
                .project_name(&name)
                .append(true)
                .build();
            ScaffoldDescription::new(opts)?.scaffold()?;
```

I get an error if the directory for the files already exists:
```bash
Select a plugin template: "basic"
🔄 Append to directory /home/bryn/git/acme…
🔄 Templating files…
Error: cannot create dir : File exists (os error 17)
```

I'd like it to create directories if they do not already exist but not overwrite files if they already exist.

**Comment by BrynCooke** [comment]
Taking a look at the underlying code I think `force` may not do what I want, it deletes the directory if it already exists.

I think if `append` is specified it should create directories if they already exist, but not overwrite files.

**Comment by bnjjj** [comment]
Fixed by https://github.com/iomentum/cargo-scaffold/commit/acf5f7f0bf4e0305c6f213e972192cb2afb0859d

**Comment by bnjjj** [comment]
Published on `0.5.0`
